### PR TITLE
Use fetch API

### PR DESCRIPTION
**Pull request by running-on-sunshine**
In this branch: 

- Refactored the following functions using jQuery AJAX with built-in fetch API and promises:
  - retrieveOrders()
  - addOrder()
  - removeOrder()

@clinturbin: Could you please help review? 🙂

### CODE REVIEW

File: scripts/main.js [hunk 1]
```diff
@@ -4,35 +4,37 @@ var orders = [];
...
-            retrieveOrders();
+    fetch(url, {
+        method: "post",
+        body: JSON.stringify(newOrder),
+        headers: {
+            "Content-Type": "application/json",
         }
+    }).then(function () {
+        retrieveOrders();
     })
```
Comment (clinturbin) [inline scripts/main.js]: semicolon

File: scripts/main.js [hunk 2]
```diff
@@ -4,35 +4,37 @@ var orders = [];
...
 var retrieveOrders = function () {
-    orders = []
-    $.ajax(url, {
-        success: function (data) {
+    var retrieveOrdersPromise = fetch(url);
+    retrieveOrdersPromise.then(function (data) {
+        orders = []
+        var ordersToJson = data.json();
+        ordersToJson.then(function (data) {
             for (key in data) {
```
Comment (clinturbin) [inline scripts/main.js]: This is a possible refactor: (including semicolons)
```
var retrieveOrders = function () {
  orders = [];
  fetch(url).then(function (data) {
    data.json().then(function (data) {
      for (key in data) {
        orders.push(data[key]);
      })
      displayOrders();
    });
  });
};
```

Comment (running-on-sunshine) [reply scripts/main.js]: Thank you for your suggestion! This really helped to clean up and shorten my retrieveOrders function! 🙂

**Comment by running-on-sunshine** [comment]
Update:

- Cleaned up retrieveOrders function as a direct result from refactoring suggestion
- Added missing semicolons 😄

@clinturbin: Could you please help review? 🙂

**Comment by clinturbin** [comment]
Looks Good!! 👍

**Pull request merged**